src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs

```diff
@@ -226,7 +226,7 @@ public void Initialize()
             this.dataCollectionPort = this.dataCollectionRequestSender.InitializeCommunication();
 
             // Warn the user that execution will wait for debugger attach.
-            this.dataCollectionProcessId = this.dataCollectionLauncher.LaunchDataCollector(null, this.GetCommandLineArguments(this.dataCollectionPort));
+            this.dataCollectionProcessId = this.dataCollectionLauncher.LaunchDataCollector(InferRunSettingsHelper.GetEnvironmentVariables(SettingsXml), this.GetCommandLineArguments(this.dataCollectionPort));
```

Comment:
IMO this is not needed, & can potentially cause issues where certain Env variables, like Profiler variables if specified in runsettings would also get applied on top of datacollector, which can potentially cause data collector to be started under profiling mode, I'm not sure how will those scenarios be handled.

We can remove this for now, & add it only when needed.

Reply:
done